### PR TITLE
Feature/diet study

### PR DESCRIPTION
**Pull request by petermason**
# Description

_Description of the feature or fix with link to the issue if existing_

## On what platform have you tested the change?

- [x] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots
image
image
image
image
image
image
image

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/features/mental-health/screens/MentalHealthModal.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ function MentalHealthModal() {
   const handleSetConsent = (consent: TMentalHealthConsent) => {
     dispatch(setConsent(consent));
     if (consent === 'LATER') {
-      dispatch(setLastPresentedDate(new Date()));
+      dispatch(setLastPresentedDate(new Date().toString()));
```
Comment (selvach) [inline src/features/mental-health/screens/MentalHealthModal.tsx:24]: What's the implication of this change?

File: src/core/state/settings/slice/index.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    setHasEmailSubscription: (state, action: PayloadAction<boolean>) => {
+      return {
+        ...state,
+        hasEmailSubscription: action.payload,
+      };
+    },
+  },
+  extraReducers: {
+    [setEmailSubscription.fulfilled]: (state, action: PayloadAction<TSubscriptionResponse>) => {
+      console.log('action payload: ', action.payload);
```
Comment (selvach) [inline src/core/state/settings/slice/index.ts:50]: Stray console log

File: src/features/dashboard/DashboardScreen.tsx [hunk 3]
```diff
@@ -74,6 +82,24 @@ export const DashboardScreen: React.FC<Props> = ({ navigation, route }) => {
     await share(shareMessage);
   };
 
+  const runCurrentFeature = () => {
+    console.log(settings.currentFeature);
```
Comment (selvach) [inline src/features/dashboard/DashboardScreen.tsx:86]: Stray console.log

File: src/core/state/diet-study/slice/index.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import { RootState } from '../../root';
+import { IDietStudy, TDietStudyConsent } from '../types';
+
+const initialState: IDietStudy = {
+  consent: undefined,
+};
+
+const dietStudySlice = createSlice({
+  name: 'MentalHealthState',
```
Comment (selvach) [inline src/core/state/diet-study/slice/index.ts:11]: dietStudySlice has MentalHealthState. 
I also don't think this is required, so we can remove the consent

File: src/features/diet-study-playback/v2/DietStudyModal.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  const [tracked, setTracked] = useState(false);
+  const dispatch = useDispatch();
+  const { goBack } = useNavigation();
+  const dietStudyVariant = startExperiment(experiments.UK_DIET_SCORE, 2);
+
+  const handleSetConsent = (consent: TDietStudyConsent) => {
+    dispatch(setDietStudyConsent(consent));
+    const date = new Date();
+    date.setDate(date.getDate() + 1);
+    dispatch(setCurrentFeature('MENTAL_HEALTH_STUDY'));
```
Comment (selvach) [inline src/features/diet-study-playback/v2/DietStudyModal.tsx:25]: Could you explain with a comment what is going on here? 
It's not obvious why you'd set the the current features to MENTAL_HEALTH_STUDY in the DietStudyModal

File: src/core/Analytics.ts [hunk 6]
```diff
@@ -20,6 +20,13 @@ const DietStudyEvents = {
   DECLINE_DIET_STUDY: 'DECLINE_DIET_STUDY',
   SIGNED_DIET_STUDY_CONSENT: 'SIGNED_DIET_STUDY_CONSENT',
   DECLINE_DIET_STUDY_CONSENT: 'DECLINE_DIET_STUDY_CONSENT',
+  DIET_STUDY_SCREEN_MODAL: 'DIET_STUDY_SCREEN_MODAL',
+  DIET_STUDY_SCREEN_START: 'DIET_STUDY_SCREEN_START',
+  DIET_STUDY_SCREEN_GLOBAL: 'DIET_STUDY_SCREEN_GLOBAL',
+  DIET_STUDY_SCREEN_GUT: 'DIET_STUDY_SCREEN_GUT',
+  DIET_STUDY_SCREEN_TRADITIONAL: 'DIET_STUDY_SCREEN_TRADITIONAL',
```
Comment (selvach) [inline src/core/Analytics.ts:27]: All screen views are tracked by default by this piece of code. 
I believe we can remove these DIET_SCREEN_* events and the bits of code that dispatches them on each screen.

**Review by selvach — COMMENTED** [review]
Looks amazing, @petermason !!

I've made some visual tweaks to bring us closer to the Figma.

Please can you check the latest commits I made. 
There's an outstanding question about a slice name and the screen track events. 

Once those a resolved, happy to approve and merge!

**Comment by selvach** [comment]
There are type errors on `DietStudyEmailModal`

File: src/features/mental-health/screens/MentalHealthModal.tsx (hunk 1 above)
Comment (petermason) [reply src/features/mental-health/screens/MentalHealthModal.tsx:24]: redux should be serializable therefore date object needs to be stored as strings

File: src/core/state/diet-study/slice/index.ts (hunk 4 above)
Comment (petermason) [reply src/core/state/diet-study/slice/index.ts]: fixed

File: src/features/diet-study-playback/v2/DietStudyModal.tsx (hunk 5 above)
Comment (petermason) [reply src/features/diet-study-playback/v2/DietStudyModal.tsx:25]: I touched on this the other morning during stand-up. Ideally this function should be more like 'setNextFeature' or 'setToNextFeature'. I've done this for now, but would like to raise a seperate ticket for scheduling features.

**Review by selvach — APPROVED** [review]
Awesome!

**Pull request merged**